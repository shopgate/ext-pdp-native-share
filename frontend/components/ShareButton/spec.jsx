import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

const mockedConfig = {
  gmdIcon: 'gmd',
  iOSIcon: 'ios',
};

jest.mock('../../helpers/getConfig', () => () => mockedConfig);
const mockedParams = {
  mockedShareParams: {
    title: 'title',
    imageURL: 'imageURL',
    deepLink: 'deepLink',
  },
};

let mockedIsIOS = true;
jest.mock('@shopgate/pwa-extension-kit', () => ({
  env: {
    isIOSTheme: () => mockedIsIOS
  }
}));

jest.mock('../../selectors/index', () => ({
  getShareParams: () => mockedParams.mockedShareParams,
}));

jest.mock('@shopgate/pwa-extension-kit/connectors', () => ({
  withPageProductId: WrappedComponent => () => <WrappedComponent productId="foo" />,
}));

/**
 * Mocked function for shareItem
 * @param {Object} mockedShareParams mocked parameters
 */
const mockedShareItem = jest.fn();
jest.mock('@shopgate/pwa-core/commands/shareItem', () => () => mockedShareItem());

describe('ShareButton', () => {
  // eslint-disable-next-line require-jsdoc
  const makeComponent = () => {
    // eslint-disable-next-line global-require
    const ShareButton = require('./index').default;
    return mount((
      <Provider store={configureStore()({})}>
        <ShareButton />
      </Provider>
    ));
  };

  it('should render when shareParams exist and handle click', () => {
    const component = makeComponent();
    expect(component.find('ShareButton').props().shareParams).toMatchObject(mockedParams.mockedShareParams);
    expect(component).toMatchSnapshot();

    component.find('ShareButton').instance().handleClick();
    expect(mockedShareItem).toHaveBeenCalled();
  });

  it('should render with gmd icon relying on theme', () => {
    mockedIsIOS = false;
    const component = makeComponent();
    expect(component.find('ShareButton').props().shareParams).toMatchObject(mockedParams.mockedShareParams);
    expect(component).toMatchSnapshot();
  });

  it('should render with ios icon relying on theme', () => {
    mockedIsIOS = true;
    const component = makeComponent();
    expect(component.find('ShareButton').props().shareParams).toMatchObject(mockedParams.mockedShareParams);
    expect(component).toMatchSnapshot();
  });

  it('should render with ios relying on config', () => {
    mockedIsIOS = false;
    mockedConfig.gmdIcon = 'ios';
    const component = makeComponent();
    expect(component.find('ShareButton').props().shareParams).toMatchObject(mockedParams.mockedShareParams);
    expect(component).toMatchSnapshot();
  });

  it('should render with gmd relying on config', () => {
    mockedIsIOS = true;
    mockedConfig.iOSIcon = 'gmd';
    const component = makeComponent();
    expect(component.find('ShareButton').props().shareParams).toMatchObject(mockedParams.mockedShareParams);
    expect(component).toMatchSnapshot();
  });

  it('should not render when deeplink is undefined', () => {
    mockedParams.mockedShareParams.deepLink = undefined;
    const component = makeComponent();
    expect(component.html()).toBe(null);
  });
});
