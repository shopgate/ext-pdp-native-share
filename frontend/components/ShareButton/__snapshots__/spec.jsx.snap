// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareButton should render when shareParams exist and handle click 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ShareButton)>
    <ShareButton
      dispatch={[Function]}
      shareItem={[Function]}
      shareParams={
        Object {
          "deepLink": "deepLink",
          "imageURL": "imageURL",
          "title": "title",
        }
      }
    >
      <button
        className="css-lard6l"
        data-test-id="shareIcon"
        type="button"
      >
        <Ripple
          className="css-zv4pyz"
          color="#000"
          disabled={false}
          fill={false}
          onClick={[Function]}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-zv4pyz"
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            <ShareIcon>
              <Icon
                className=""
                color={null}
                content="
    <g>
      <path d=\\"M15.3,6.9c-0.5,0-0.9,0.4-0.9,0.9s0.4,0.9,0.9,0.9h2.1v10.8H6.6V8.7h2.1c0.5,0,0.9-0.4,0.9-0.9S9.2,6.9,8.7,6.9H4.8v14.4 h14.4V6.9H15.3z\\"/>
      <path d=\\"M9.9,5.8l1.4-1.4v8.9c0,0.4,0.4,0.8,0.8,0.8c0.4,0,0.8-0.4,0.8-0.8V4.5l1.4,1.4c0.2,0.2,0.4,0.2,0.6,0.2s0.4-0.1,0.6-0.2 c0.3-0.3,0.3-0.8,0-1.1L12,1.5L8.7,4.7C8.4,5,8.4,5.5,8.7,5.8C9,6.1,9.5,6.1,9.9,5.8z\\"/>
  </g>
  "
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "
    <g>
      <path d=\\"M15.3,6.9c-0.5,0-0.9,0.4-0.9,0.9s0.4,0.9,0.9,0.9h2.1v10.8H6.6V8.7h2.1c0.5,0,0.9-0.4,0.9-0.9S9.2,6.9,8.7,6.9H4.8v14.4 h14.4V6.9H15.3z\\"/>
      <path d=\\"M9.9,5.8l1.4-1.4v8.9c0,0.4,0.4,0.8,0.8,0.8c0.4,0,0.8-0.4,0.8-0.8V4.5l1.4,1.4c0.2,0.2,0.4,0.2,0.6,0.2s0.4-0.1,0.6-0.2 c0.3-0.3,0.3-0.8,0-1.1L12,1.5L8.7,4.7C8.4,5,8.4,5.5,8.7,5.8C9,6.1,9.5,6.1,9.9,5.8z\\"/>
  </g>
  ",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </ShareIcon>
          </div>
        </Ripple>
      </button>
    </ShareButton>
  </Connect(ShareButton)>
</Provider>
`;

exports[`ShareButton should render with gmd icon relying on theme 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ShareButton)>
    <ShareButton
      dispatch={[Function]}
      shareItem={[Function]}
      shareParams={
        Object {
          "deepLink": "deepLink",
          "imageURL": "imageURL",
          "title": "title",
        }
      }
    >
      <button
        className="css-1sj1wok"
        data-test-id="shareIcon"
        type="button"
      >
        <Ripple
          className="css-ied7gv"
          color="#000"
          disabled={false}
          fill={false}
          onClick={[Function]}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-ied7gv"
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            <ShareIcon>
              <Icon
                className=""
                color={null}
                content="<path d=\\"M18 16.1364C17.24 16.1364 16.56 16.4376 16.04 16.9095L8.91 12.7428C8.96 12.5119 9 12.281 9 12.04C9 11.7991 8.96 11.5682 8.91 11.3372L15.96 7.21072C16.5 7.71273 17.21 8.02397 18 8.02397C19.66 8.02397 21 6.67859 21 5.01193C21 3.34526 19.66 1.99988 18 1.99988C16.34 1.99988 15 3.34526 15 5.01193C15 5.25289 15.04 5.48381 15.09 5.71474L8.04 9.84124C7.5 9.33924 6.79 9.02799 6 9.02799C4.34 9.02799 3 10.3734 3 12.04C3 13.7067 4.34 15.0521 6 15.0521C6.79 15.0521 7.5 14.7408 8.04 14.2388L15.16 18.4155C15.11 18.6264 15.08 18.8473 15.08 19.0682C15.08 20.6846 16.39 21.9999 18 21.9999C19.61 21.9999 20.92 20.6846 20.92 19.0682C20.92 17.4517 19.61 16.1364 18 16.1364Z\\"/>"
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M18 16.1364C17.24 16.1364 16.56 16.4376 16.04 16.9095L8.91 12.7428C8.96 12.5119 9 12.281 9 12.04C9 11.7991 8.96 11.5682 8.91 11.3372L15.96 7.21072C16.5 7.71273 17.21 8.02397 18 8.02397C19.66 8.02397 21 6.67859 21 5.01193C21 3.34526 19.66 1.99988 18 1.99988C16.34 1.99988 15 3.34526 15 5.01193C15 5.25289 15.04 5.48381 15.09 5.71474L8.04 9.84124C7.5 9.33924 6.79 9.02799 6 9.02799C4.34 9.02799 3 10.3734 3 12.04C3 13.7067 4.34 15.0521 6 15.0521C6.79 15.0521 7.5 14.7408 8.04 14.2388L15.16 18.4155C15.11 18.6264 15.08 18.8473 15.08 19.0682C15.08 20.6846 16.39 21.9999 18 21.9999C19.61 21.9999 20.92 20.6846 20.92 19.0682C20.92 17.4517 19.61 16.1364 18 16.1364Z\\"/>",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </ShareIcon>
          </div>
        </Ripple>
      </button>
    </ShareButton>
  </Connect(ShareButton)>
</Provider>
`;

exports[`ShareButton should render with gmd relying on config 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ShareButton)>
    <ShareButton
      dispatch={[Function]}
      shareItem={[Function]}
      shareParams={
        Object {
          "deepLink": "deepLink",
          "imageURL": "imageURL",
          "title": "title",
        }
      }
    >
      <button
        className="css-10dnx6t"
        data-test-id="shareIcon"
        type="button"
      >
        <Ripple
          className="css-zv4pyz"
          color="#000"
          disabled={false}
          fill={false}
          onClick={[Function]}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-zv4pyz"
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            <ShareIcon>
              <Icon
                className=""
                color={null}
                content="<path d=\\"M18 16.1364C17.24 16.1364 16.56 16.4376 16.04 16.9095L8.91 12.7428C8.96 12.5119 9 12.281 9 12.04C9 11.7991 8.96 11.5682 8.91 11.3372L15.96 7.21072C16.5 7.71273 17.21 8.02397 18 8.02397C19.66 8.02397 21 6.67859 21 5.01193C21 3.34526 19.66 1.99988 18 1.99988C16.34 1.99988 15 3.34526 15 5.01193C15 5.25289 15.04 5.48381 15.09 5.71474L8.04 9.84124C7.5 9.33924 6.79 9.02799 6 9.02799C4.34 9.02799 3 10.3734 3 12.04C3 13.7067 4.34 15.0521 6 15.0521C6.79 15.0521 7.5 14.7408 8.04 14.2388L15.16 18.4155C15.11 18.6264 15.08 18.8473 15.08 19.0682C15.08 20.6846 16.39 21.9999 18 21.9999C19.61 21.9999 20.92 20.6846 20.92 19.0682C20.92 17.4517 19.61 16.1364 18 16.1364Z\\"/>"
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path d=\\"M18 16.1364C17.24 16.1364 16.56 16.4376 16.04 16.9095L8.91 12.7428C8.96 12.5119 9 12.281 9 12.04C9 11.7991 8.96 11.5682 8.91 11.3372L15.96 7.21072C16.5 7.71273 17.21 8.02397 18 8.02397C19.66 8.02397 21 6.67859 21 5.01193C21 3.34526 19.66 1.99988 18 1.99988C16.34 1.99988 15 3.34526 15 5.01193C15 5.25289 15.04 5.48381 15.09 5.71474L8.04 9.84124C7.5 9.33924 6.79 9.02799 6 9.02799C4.34 9.02799 3 10.3734 3 12.04C3 13.7067 4.34 15.0521 6 15.0521C6.79 15.0521 7.5 14.7408 8.04 14.2388L15.16 18.4155C15.11 18.6264 15.08 18.8473 15.08 19.0682C15.08 20.6846 16.39 21.9999 18 21.9999C19.61 21.9999 20.92 20.6846 20.92 19.0682C20.92 17.4517 19.61 16.1364 18 16.1364Z\\"/>",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </ShareIcon>
          </div>
        </Ripple>
      </button>
    </ShareButton>
  </Connect(ShareButton)>
</Provider>
`;

exports[`ShareButton should render with ios icon relying on theme 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ShareButton)>
    <ShareButton
      dispatch={[Function]}
      shareItem={[Function]}
      shareParams={
        Object {
          "deepLink": "deepLink",
          "imageURL": "imageURL",
          "title": "title",
        }
      }
    >
      <button
        className="css-lard6l"
        data-test-id="shareIcon"
        type="button"
      >
        <Ripple
          className="css-zv4pyz"
          color="#000"
          disabled={false}
          fill={false}
          onClick={[Function]}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-zv4pyz"
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            <ShareIcon>
              <Icon
                className=""
                color={null}
                content="
    <g>
      <path d=\\"M15.3,6.9c-0.5,0-0.9,0.4-0.9,0.9s0.4,0.9,0.9,0.9h2.1v10.8H6.6V8.7h2.1c0.5,0,0.9-0.4,0.9-0.9S9.2,6.9,8.7,6.9H4.8v14.4 h14.4V6.9H15.3z\\"/>
      <path d=\\"M9.9,5.8l1.4-1.4v8.9c0,0.4,0.4,0.8,0.8,0.8c0.4,0,0.8-0.4,0.8-0.8V4.5l1.4,1.4c0.2,0.2,0.4,0.2,0.6,0.2s0.4-0.1,0.6-0.2 c0.3-0.3,0.3-0.8,0-1.1L12,1.5L8.7,4.7C8.4,5,8.4,5.5,8.7,5.8C9,6.1,9.5,6.1,9.9,5.8z\\"/>
  </g>
  "
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "
    <g>
      <path d=\\"M15.3,6.9c-0.5,0-0.9,0.4-0.9,0.9s0.4,0.9,0.9,0.9h2.1v10.8H6.6V8.7h2.1c0.5,0,0.9-0.4,0.9-0.9S9.2,6.9,8.7,6.9H4.8v14.4 h14.4V6.9H15.3z\\"/>
      <path d=\\"M9.9,5.8l1.4-1.4v8.9c0,0.4,0.4,0.8,0.8,0.8c0.4,0,0.8-0.4,0.8-0.8V4.5l1.4,1.4c0.2,0.2,0.4,0.2,0.6,0.2s0.4-0.1,0.6-0.2 c0.3-0.3,0.3-0.8,0-1.1L12,1.5L8.7,4.7C8.4,5,8.4,5.5,8.7,5.8C9,6.1,9.5,6.1,9.9,5.8z\\"/>
  </g>
  ",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </ShareIcon>
          </div>
        </Ripple>
      </button>
    </ShareButton>
  </Connect(ShareButton)>
</Provider>
`;

exports[`ShareButton should render with ios relying on config 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ShareButton)>
    <ShareButton
      dispatch={[Function]}
      shareItem={[Function]}
      shareParams={
        Object {
          "deepLink": "deepLink",
          "imageURL": "imageURL",
          "title": "title",
        }
      }
    >
      <button
        className="css-lard6l"
        data-test-id="shareIcon"
        type="button"
      >
        <Ripple
          className="css-ied7gv"
          color="#000"
          disabled={false}
          fill={false}
          onClick={[Function]}
          onComplete={[Function]}
          overflow={false}
          size={null}
        >
          <div
            aria-hidden={true}
            className="css-ied7gv"
            data-test-id="Ripple"
            onClick={[Function]}
            style={null}
          >
            <ShareIcon>
              <Icon
                className=""
                color={null}
                content="
    <g>
      <path d=\\"M15.3,6.9c-0.5,0-0.9,0.4-0.9,0.9s0.4,0.9,0.9,0.9h2.1v10.8H6.6V8.7h2.1c0.5,0,0.9-0.4,0.9-0.9S9.2,6.9,8.7,6.9H4.8v14.4 h14.4V6.9H15.3z\\"/>
      <path d=\\"M9.9,5.8l1.4-1.4v8.9c0,0.4,0.4,0.8,0.8,0.8c0.4,0,0.8-0.4,0.8-0.8V4.5l1.4,1.4c0.2,0.2,0.4,0.2,0.6,0.2s0.4-0.1,0.6-0.2 c0.3-0.3,0.3-0.8,0-1.1L12,1.5L8.7,4.7C8.4,5,8.4,5.5,8.7,5.8C9,6.1,9.5,6.1,9.9,5.8z\\"/>
  </g>
  "
                size="inherit"
                viewBox="0 0 24 24"
              >
                <svg
                  className="css-yys9hb "
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "
    <g>
      <path d=\\"M15.3,6.9c-0.5,0-0.9,0.4-0.9,0.9s0.4,0.9,0.9,0.9h2.1v10.8H6.6V8.7h2.1c0.5,0,0.9-0.4,0.9-0.9S9.2,6.9,8.7,6.9H4.8v14.4 h14.4V6.9H15.3z\\"/>
      <path d=\\"M9.9,5.8l1.4-1.4v8.9c0,0.4,0.4,0.8,0.8,0.8c0.4,0,0.8-0.4,0.8-0.8V4.5l1.4,1.4c0.2,0.2,0.4,0.2,0.6,0.2s0.4-0.1,0.6-0.2 c0.3-0.3,0.3-0.8,0-1.1L12,1.5L8.7,4.7C8.4,5,8.4,5.5,8.7,5.8C9,6.1,9.5,6.1,9.9,5.8z\\"/>
  </g>
  ",
                    }
                  }
                  style={
                    Object {
                      "fill": null,
                      "fontSize": "inherit",
                    }
                  }
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </Icon>
            </ShareIcon>
          </div>
        </Ripple>
      </button>
    </ShareButton>
  </Connect(ShareButton)>
</Provider>
`;
